### PR TITLE
Component cleanup

### PR DESCRIPTION
**Pull request by mathisonian**
This is a set of fixes that I've done while trying to make a real world example using the updated v2 modules. It mostly deals with fixing some bugs in the scroller components that were recently added. 

It also removes the wrapping `<span>` tag from the `Wrap` component in favor of just returning an array.

### CODE REVIEW

**Comment by mathisonian** [comment]
@rreusser this uses some of your example code to get a fullscreen feature component that it is all CSS and works with SSR ✨

**Comment by rreusser** [comment]
@mathisonian Awesome! I haven't forgotten about idyll. I'll have to give this a spin. 🎉

File: packages/idyll-components/src/feature.js [hunk 1]
```diff
@@ -64,12 +66,49 @@ class Feature extends React.PureComponent {
...
+    const unwrapped = this.unwrapChildren();
+    return React.Children.toArray(this.props.children).filter((_, i) => {
+      const c = unwrapped[i];
+      if (!c.type) {
+        return true;
+      }
+      return (!c.type.name || c.type.name.toLowerCase() !== 'content') && !(c.type.prototype instanceof Content);
+    });
+  }
+
```
Comment (bclinkinbeard) [inline packages/idyll-components/src/feature.js]: Nit, but combining these functions would be nice. Could return an array or object perfect for destructuring.

File: packages/idyll-components/src/panel.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ class Panel extends React.PureComponent {
   }
 
   render() {
-    return <div className="panel"  {...this.props} />;
+    const { updateProps, hasError, ...props } = this.props;
+    return <div className="panel"  {...props} />;
```
Comment (bclinkinbeard) [inline packages/idyll-components/src/panel.js]: Double space snuck in

File: packages/idyll-components/src/waypoint.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ class Waypoint extends React.PureComponent {
   }
 
   render() {
-    return <Screen direction="row"  {...this.props} />;
+    return <Screen align="center" direction="row"  {...this.props} />;
```
Comment (bclinkinbeard) [inline packages/idyll-components/src/waypoint.js]: Another double space

File: packages/idyll-document/src/index.js [hunk 4]
```diff
@@ -45,10 +45,12 @@ const getRefs = () => {
   return refs;
 };
 
+let wrapperKey = 0;
```
Comment (bclinkinbeard) [inline packages/idyll-document/src/index.js]: Does this get updated somewhere?

**Pull request merged**